ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -47,12 +56,34 @@ cpdef pyframe_to_frames(frame, max_nframes):
     :param frame: The frame object to serialize.
     :param max_nframes: The maximum number of frames to return.
     :return: The serialized frames and the number of frames present in the original traceback."""
+    # DEV: There are reports that Python 3.11 returns non-frame objects when
+    # retrieving frame objects and doing stack unwinding. If we detect a
+    # non-frame object we log a warning. This until we can come to the bottom
+    # of the issue.
+    if not isinstance(frame, FrameType):
+        log.warning(
+            "Got object of type '%s' instead of a frame object for the top frame of a thread", type(frame).__name__
+        )
+        return [], 0
```

Comment:
I want to capture a different warning message in case the problem is already at the top span. This will be the indication that we got an "alien" frame when collecting the top frame from each thread, as opposed to during the process of stack unwinding performed here

Reply:
Got it, that makes sense to me 👍